perf/src/main.rs

```diff
@@ -0,0 +1,118 @@
+use std::net::{IpAddr, Ipv6Addr, SocketAddr};
+
+use anyhow::{Context, Result};
+use futures::StreamExt;
+use tracing::{debug, error, info};
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::fmt::init();
+
+    let cert = rcgen::generate_simple_self_signed(vec!["localhost".into()]).unwrap();
+    let key = quinn::PrivateKey::from_der(&cert.serialize_private_key_der()).unwrap();
+    let cert = quinn::Certificate::from_der(&cert.serialize_der().unwrap()).unwrap();
+
+    let mut server_config = quinn::ServerConfigBuilder::default();
+    server_config
+        .certificate(quinn::CertificateChain::from_certs(vec![cert.clone()]), key)
+        .unwrap();
+    server_config.protocols(&[b"perf"]);
+
+    let server_config = server_config.build();
+
+    let mut endpoint = quinn::EndpointBuilder::default();
+    endpoint.listen(server_config);
+
+    let (endpoint, mut incoming) = endpoint
+        .bind(&SocketAddr::new(IpAddr::V6(Ipv6Addr::UNSPECIFIED), 0))
+        .unwrap();
+
+    info!("listening on {}", endpoint.local_addr().unwrap());
+
+    while let Some(handshake) = incoming.next().await {
+        tokio::spawn(async move {
+            if let Err(e) = accept(handshake).await {
+                error!("connection lost: {}", e);
+            }
+        });
+    }
+}
+
+async fn accept(handshake: quinn::Connecting) -> Result<()> {
+    let quinn::NewConnection {
+        uni_streams,
+        bi_streams,
+        connection,
+        ..
+    } = handshake.await.context("handshake failed")?;
+    debug!("{} connected", connection.remote_address());
+    tokio::select! {
+        x = drive_uni(connection, uni_streams) => x?,
+        x = drive_bi(bi_streams) => x?,
+    }
+    Ok(())
+}
+
+async fn drive_uni(
+    connection: quinn::Connection,
+    mut streams: quinn::IncomingUniStreams,
+) -> Result<()> {
+    while let Some(stream) = streams.next().await {
+        let stream = stream?;
+        let connection = connection.clone();
+        tokio::spawn(async move {
+            if let Err(e) = accept_uni(connection, stream).await {
```

Comment:
I would rather call it `handle_uni`. `accept` sounds like it would wait for a stream. However that already had happened. Same for `accept_bi`